### PR TITLE
Remove dependency on Guava

### LINKED ISSUE #38: [FR] Remove dependency on Guava
Is it possibly to remove the dependency on Guava so I don't need to multidex my builds?

### PR DESCRIPTION
**Pull request by felipecsl**
~~Still a work in progress, but should be done soon, still a few usages of `Splitter` and `Ordering` left.~~

Fixes #38

### CODE REVIEW

**Comment by christopherperry** [comment]
Thank you @felipecsl

File: okreplay-core/build.gradle [hunk 1]
```diff
@@ -7,9 +7,9 @@ sourceCompatibility = 1.7
 targetCompatibility = 1.7
 
 dependencies {
-  compile dep.guava
   compile dep.snakeYaml
   compile dep.okhttp
+  compile dep.jsr305
```
Comment (jaredsburrows) [inline okreplay-core/build.gradle]: 👍

File: okreplay-core/src/main/java/okreplay/Absent.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  @Override public String toString() {
+    return "Optional.absent()";
+  }
+
+  private Object readResolve() {
+    return INSTANCE;
+  }
+
+  private static final long serialVersionUID = 0;
```
Comment (jaredsburrows) [inline okreplay-core/src/main/java/okreplay/Absent.java]: `0L`?

File: okreplay-core/src/main/java/okreplay/Optional.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+import java.io.Serializable;
+import java.util.Set;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import static okreplay.Util.checkNotNull;
+
+abstract class Optional<T> implements Serializable {
```
Comment (jaredsburrows) [inline okreplay-core/src/main/java/okreplay/Optional.java]: Maybe use a library for this? It could be easier: https://github.com/memoizr/retro-optional.

**Comment by jaredsburrows** [comment]
This is a good idea. Smaller dependency when adding this to the `androidTestCompile` configuration.

**Comment by felipecsl** [comment]
@jaredsburrows this module is actually for the `debugCompile` dependency, so it has a big impact on debug builds 👍

**Comment by christopherperry** [comment]
@felipecsl That's why I almost died when I saw it haha. Thanks for the quick turnaround on this one.

File: okreplay-core/src/main/java/okreplay/Optional.java (hunk 3 above)
Comment (felipecsl) [inline okreplay-core/src/main/java/okreplay/Optional.java]: That would work too, but it's so simple that's probably not worth it

**Comment by jaredsburrows** [comment]
LGTM

**Comment by felipecsl** [comment]
thanks for the review @jaredsburrows

**Pull request merged**